src/gui/maptools/qgsmaptoolidentify.h

```diff
@@ -246,14 +246,16 @@ class GUI_EXPORT QgsMapToolIdentify : public QgsMapTool
      *
      * This is useful when the identification is triggered by some other piece of GUI like a 3D map view
      * and some properties like search radius need to be adjusted so that identification returns correct
-     * results. Currently only search radius may be overridden.
-     *
+     * results.
      * When the custom identification has finished, restoreCanvasPropertiesOverrides() should
      * be called to erase any overrides.
+     *
+     * \param searchRadiusMapUnits The overridden search radius in map units
+     * \param skip3DLayers Optional override to skip identify results from layers that have a 3d renderer set (since QGIS 3.42)
      * \see restoreCanvasPropertiesOverrides()
      * \since QGIS 3.4
      */
-    void setCanvasPropertiesOverrides( double searchRadiusMapUnits );
+    void setCanvasPropertiesOverrides( double searchRadiusMapUnits, bool skip3DLayers = true );
```

Comment:
Instead of a bool argument, could you use a flag enum instead? That's a little more future proof and avoids the risk of multiple `true, false, true` style arguments (and also the risk of accidental automatic int/double to bool conversion if arguments are specified in the wrong order)

Reply:
> wouldn't it be better to replace parameters with a variant map, or better a simple CanvasPropertiesOverrides struct? The flag can only handle bool-ish parameters

Not a variant map, cos that's extremely fragile -- but yes, a CanvasPropertiesOverrides struct would be even better. It's just more work vs a enum :)

Reply:
@nyalldawson I've added a `IdentifyProperties` struct (the properties will not all be tied to canvas, so skipped it from the name) and new methods to override/restore defaults.